### DISCUSSION TITLE
Conditional property exclusion on serialization

### DISCUSSION THREAD

**Discussion by innocenzi** [spatie/laravel-data#111] [Q&A]
Hey,

I'm wondering if there is a nice way to conditionally exclude (or include) properties of a DTO.

In my case, I'm using `laravel-data` together with Inertia, and I'd like to not expose certain properties depending on the authorization level of users. For instance, I don't want `id` to be present most of the times, but I do want them in the admin panel. 

Is there any way to do that without creating another DTO? I couldn't find anything on the docs. I don't think lazy properties solve the use case because we can force them via URL (and I didn't see a way to disable that), and I'd like to not have to redefine a `from*` method. 

Ideally, something simple like:

```php
public function excludeWhen() {
    return [
        'id' => !auth()->user()->is_admin
    ];
}
```

**Comment by innocenzi** [comment]
Looks like https://github.com/spatie/laravel-data/discussions/5 would be similar

**Comment by rubenvanassche** [comment]
Hi @innocenzi,

At the moment there's no support for this. I think some people are looking into it but I'm not sure. Currently working on data v2 so I've added it on a maybe list(https://github.com/spatie/laravel-data/discussions/84) but I really want to get this version released without constantly adding new features 😄 . Feel free to send a PR adding this feature.

**Reply by innocenzi** [reply]
Thanks for the answer!
